jax/_src/state/indexing.py

```diff
@@ -53,7 +53,7 @@ def from_slice(cls, slc: slice, size: int) -> Slice:
     start, stop, step = slc.indices(size)
     if step != 1:
       raise ValueError(f"slice must have a step of 1 (found: {step})")
-    return cls(start, stop - start)
+    return cls(start, max(stop - start, 0))
```

Comment:
Please use `core.max_dim` instead of `max` to ensure this code works with dynamic shapes.

Reply:
Actually, this code is not ready for polymorphic shapes, so you can submit as is.